### PR TITLE
Require Senior Delegate if delegate_status allows it, fixes #2933.

### LINKED ISSUE #2933: If delegate_status_allows_senior_delegate a senior_delegate should be required
The title is self explanatory I think. I came when around this while working on #1046, and actually I started fixing it, until I saw how many tests it broke :P 

Adding such a validation is quite easy:
```ruby
validate :must_have_senior_delegate
def must_have_senior_delegate
  if User.delegate_status_allows_senior_delegate(delegate_status) && !senior_delegate
    errors.add(:senior_delegate, I18n.t('users.errors.must_be_present'))
  end
end
```

And the test too:
```ruby
it "requires senior delegate if delegate status allows it" do
  delegate = FactoryBot.create :delegate

  delegate.senior_delegate = nil
  expect(delegate).to be_invalid_with_errors(["must be present"])
end
```

Also, a change is needed on the en.yml file:
```ruby
must_be_present: "must be present"
```

I think the issue is that the factory by default generates a `delegate` without a `senior_delegate`, but I'm not sure how to change that or if that will break another awful lot of tests. Anyway, I felt that this deserved a separate issue and PR, so here I am :)

### PR DESCRIPTION
**Pull request by AlbertoPdRF**
WIP: there are a lot of test fails that I need to figure out how to fix yet, but I'm creating the PR now just in case I'm doing something wrong and I need to fix it.

### CODE REVIEW

File: WcaOnRails/app/models/user.rb [hunk 1]
```diff
@@ -298,6 +298,13 @@ def senior_delegate_presence
     end
   end
 
+  validate :must_have_senior_delegate
+  def must_have_senior_delegate
+    if User.delegate_status_allows_senior_delegate(delegate_status) && !senior_delegate
+      errors.add(:senior_delegate, I18n.t('users.errors.must_be_present'))
+    end
+  end
+
```
Comment (jonatanklosko) [inline WcaOnRails/app/models/user.rb]: I believe you can use Rails build-in validation, which wouldn't require new translation phrase.
Something along the lines of:
```rb
validates :senior_delegate, presence: true, if: -> { User.delegate_status_allows_senior_delegate(delegate_status) && !senior_delegate }
```
This should result in the same error message, but already translated =)

File: WcaOnRails/spec/models/user_spec.rb [hunk 2]
```diff
@@ -80,6 +80,13 @@
     expect(delegate).to be_valid
   end
 
+  it "requires senior delegate if delegate status allows it" do
+    delegate = FactoryBot.create :delegate
+
+    delegate.senior_delegate = nil
+    expect(delegate).to be_invalid_with_errors(senior_delegate: ["must be present"])
+  end
```
Comment (jonatanklosko) [inline WcaOnRails/spec/models/user_spec.rb]: You can specify attributes during creation to simplify this:
```rb
it "requires senior delegate if delegate status allows it" do
  delegate = FactoryBot.build :delegate, senior_delegate: nil
  expect(delegate).to be_invalid_with_errors(senior_delegate: ["must be present"])
end
```
Also note that I used `build` instead` of `create`. It's fine both ways, but given we know that creation should fail, there's no reason to try saving it to the database.

File: WcaOnRails/app/models/user.rb (hunk 1 above)
Comment (AlbertoPdRF) [reply WcaOnRails/app/models/user.rb]: That's really neat!

Comment (jfly) [reply WcaOnRails/app/models/user.rb]: @AlbertoPdRF, with the work you're doing in this PR, should we rename `delegate_status_allows_senior_delegate` to `delegate_status_requires_senior_delegate`?

Comment (AlbertoPdRF) [reply WcaOnRails/app/models/user.rb]: Sounds like a good change to me, should I proceed?

Comment (jfly) [reply WcaOnRails/app/models/user.rb]: Go for it!

File: WcaOnRails/spec/models/user_spec.rb (hunk 2 above)
Comment (AlbertoPdRF) [reply WcaOnRails/spec/models/user_spec.rb]: Noted, thanks!

**Pull request merged**